packages/hardhat-zksync-solc/src/index.ts

```diff
@@ -30,6 +48,13 @@ extendEnvironment((hre) => {
         hre.config.paths.artifacts = artifactsPath;
         hre.config.paths.cache = cachePath;
         (hre as any).artifacts = new Artifacts(artifactsPath);
+
+        // If solidity optimizer is not enabled, the libraries are not inlined and
+        // we have to manually pass them into zksolc. So for now we force the optimization.
+        hre.config.solidity.compilers.forEach((compiler) => {
```

Comment:
Maybe we should display an error whenever a user tries to set optimizer as `false`? So that everything is explicit?

Reply:
I think a warning is better. I'll do it in a separate PR.